### ISSUE TITLE
couchbase-lite-java-javascript:unspecified depends on libraries but is not a library itself

### ISSUE THREAD

**Issue by andreibaranouski** [couchbase/couchbase-lite-android-liteserv#29]
http://qa.hq.northscale.net/job/centos_x64-mobile-sync-gateway-tests-android_cb_bucket_shadowing/46/console

```
Cloning into 'couchbase-lite-android-liteserv'...
Submodule 'libraries/couchbase-lite-android' (https://github.com/couchbase/couchbase-lite-android.git) registered for path 'libraries/couchbase-lite-android'
Submodule 'libraries/couchbase-lite-java-core' (https://github.com/couchbase/couchbase-lite-java-core.git) registered for path 'libraries/couchbase-lite-java-core'
Submodule 'libraries/couchbase-lite-java-forestdb' (https://github.com/couchbaselabs/couchbase-lite-java-forestdb) registered for path 'libraries/couchbase-lite-java-forestdb'
Submodule 'libraries/couchbase-lite-java-javascript' (https://github.com/couchbase/couchbase-lite-java-javascript.git) registered for path 'libraries/couchbase-lite-java-javascript'
Submodule 'libraries/couchbase-lite-java-listener' (https://github.com/couchbase/couchbase-lite-java-listener.git) registered for path 'libraries/couchbase-lite-java-listener'
Submodule 'libraries/couchbase-lite-java-sqlcipher' (https://github.com/couchbaselabs/couchbase-lite-java-sqlcipher) registered for path 'libraries/couchbase-lite-java-sqlcipher'
Cloning into 'libraries/couchbase-lite-android'...
Submodule path 'libraries/couchbase-lite-android': checked out 'd2e6ff09aa0b5ef902cae1b958e0b78a07d4d2cd'
Submodule 'libraries/couchbase-lite-java-core' (https://github.com/couchbase/couchbase-lite-java-core.git) registered for path 'libraries/couchbase-lite-java-core'
Submodule 'libraries/couchbase-lite-java-forestdb' (https://github.com/couchbaselabs/couchbase-lite-java-forestdb.git) registered for path 'libraries/couchbase-lite-java-forestdb'
Submodule 'libraries/couchbase-lite-java-sqlcipher' (https://github.com/couchbaselabs/couchbase-lite-java-sqlcipher.git) registered for path 'libraries/couchbase-lite-java-sqlcipher'
Cloning into 'libraries/couchbase-lite-java-core'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-core': checked out '35dc47568e81a22f8432c70506aba81505cdbd62'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-core/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'libraries/couchbase-lite-java-forestdb'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-forestdb': checked out 'efd2b7d5ed70073f14855d504fd266f2853545f4'
Submodule 'libraries/couchbase-lite-java-core' (https://github.com/couchbase/couchbase-lite-java-core.git) registered for path 'libraries/couchbase-lite-java-core'
Submodule 'vendor/cbforest' (https://github.com/couchbaselabs/cbforest.git) registered for path 'vendor/cbforest'
Cloning into 'libraries/couchbase-lite-java-core'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/libraries/couchbase-lite-java-core': checked out '35dc47568e81a22f8432c70506aba81505cdbd62'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/libraries/couchbase-lite-java-core/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'vendor/cbforest'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/vendor/cbforest': checked out 'b691c62cbf63e24902cb9fcb40b6bc762e5550cd'
Submodule 'vendor/forestdb' (https://github.com/couchbaselabs/forestdb.git) registered for path 'vendor/forestdb'
Submodule 'vendor/sqlite3-unicodesn' (git://github.com/snej/sqlite3-unicodesn.git) registered for path 'vendor/sqlite3-unicodesn'
Cloning into 'vendor/forestdb'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/vendor/cbforest/vendor/forestdb': checked out 'ec5f905406664acfd251efc5df1a7261db1b5884'
Cloning into 'vendor/sqlite3-unicodesn'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-forestdb/vendor/cbforest/vendor/sqlite3-unicodesn': checked out 'ddee1b572f320c55176f3db017bebe0716cfeca4'
Cloning into 'libraries/couchbase-lite-java-sqlcipher'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-sqlcipher': checked out '61703d656eec2cb530d8ecb8da5e52303a16c203'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Submodule 'vendor/sqlcipher' (https://github.com/sqlcipher/sqlcipher.git) registered for path 'vendor/sqlcipher'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-sqlcipher/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'vendor/sqlcipher'...
Submodule path 'libraries/couchbase-lite-android/libraries/couchbase-lite-java-sqlcipher/vendor/sqlcipher': checked out 'c01b94fb6a6b54e00b9839e7314893f72c9dab22'
Cloning into 'libraries/couchbase-lite-java-core'...
Submodule path 'libraries/couchbase-lite-java-core': checked out '35dc47568e81a22f8432c70506aba81505cdbd62'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-java-core/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'libraries/couchbase-lite-java-forestdb'...
Submodule path 'libraries/couchbase-lite-java-forestdb': checked out 'efd2b7d5ed70073f14855d504fd266f2853545f4'
Submodule 'libraries/couchbase-lite-java-core' (https://github.com/couchbase/couchbase-lite-java-core.git) registered for path 'libraries/couchbase-lite-java-core'
Submodule 'vendor/cbforest' (https://github.com/couchbaselabs/cbforest.git) registered for path 'vendor/cbforest'
Cloning into 'libraries/couchbase-lite-java-core'...
Submodule path 'libraries/couchbase-lite-java-forestdb/libraries/couchbase-lite-java-core': checked out '35dc47568e81a22f8432c70506aba81505cdbd62'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-java-forestdb/libraries/couchbase-lite-java-core/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'vendor/cbforest'...
Submodule path 'libraries/couchbase-lite-java-forestdb/vendor/cbforest': checked out 'b691c62cbf63e24902cb9fcb40b6bc762e5550cd'
Submodule 'vendor/forestdb' (https://github.com/couchbaselabs/forestdb.git) registered for path 'vendor/forestdb'
Submodule 'vendor/sqlite3-unicodesn' (git://github.com/snej/sqlite3-unicodesn.git) registered for path 'vendor/sqlite3-unicodesn'
Cloning into 'vendor/forestdb'...
Submodule path 'libraries/couchbase-lite-java-forestdb/vendor/cbforest/vendor/forestdb': checked out 'ec5f905406664acfd251efc5df1a7261db1b5884'
Cloning into 'vendor/sqlite3-unicodesn'...
Submodule path 'libraries/couchbase-lite-java-forestdb/vendor/cbforest/vendor/sqlite3-unicodesn': checked out 'ddee1b572f320c55176f3db017bebe0716cfeca4'
Cloning into 'libraries/couchbase-lite-java-javascript'...
Submodule path 'libraries/couchbase-lite-java-javascript': checked out '7655951282df0a367cb0e8b9768c5160487d3d4c'
Submodule 'libraries/couchbase-lite-java-core' (https://github.com/couchbase/couchbase-lite-java-core.git) registered for path 'libraries/couchbase-lite-java-core'
Cloning into 'libraries/couchbase-lite-java-core'...
Submodule path 'libraries/couchbase-lite-java-javascript/libraries/couchbase-lite-java-core': checked out '35dc47568e81a22f8432c70506aba81505cdbd62'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-java-javascript/libraries/couchbase-lite-java-core/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'libraries/couchbase-lite-java-listener'...
Submodule path 'libraries/couchbase-lite-java-listener': checked out 'b043b611e72aff0d5e2d7cddbec207b73d9071ca'
Submodule 'libraries/couchbase-lite-java-core' (https://github.com/couchbase/couchbase-lite-java-core.git) registered for path 'libraries/couchbase-lite-java-core'
Cloning into 'libraries/couchbase-lite-java-core'...
Submodule path 'libraries/couchbase-lite-java-listener/libraries/couchbase-lite-java-core': checked out '35dc47568e81a22f8432c70506aba81505cdbd62'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-java-listener/libraries/couchbase-lite-java-core/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'libraries/couchbase-lite-java-sqlcipher'...
Submodule path 'libraries/couchbase-lite-java-sqlcipher': checked out '61703d656eec2cb530d8ecb8da5e52303a16c203'
Submodule 'vendor/libtomcrypt' (https://github.com/libtom/libtomcrypt.git) registered for path 'vendor/libtomcrypt'
Submodule 'vendor/sqlcipher' (https://github.com/sqlcipher/sqlcipher.git) registered for path 'vendor/sqlcipher'
Cloning into 'vendor/libtomcrypt'...
Submodule path 'libraries/couchbase-lite-java-sqlcipher/vendor/libtomcrypt': checked out 'bbc52b9e1bf4b22ac4616e667b06d217c6ab004e'
Cloning into 'vendor/sqlcipher'...
Submodule path 'libraries/couchbase-lite-java-sqlcipher/vendor/sqlcipher': checked out 'c01b94fb6a6b54e00b9839e7314893f72c9dab22'



./run_android_liteserv.sh 8081
[couchbase-lite-java-core] No build 'spec' defined, use 'android' by default
[couchbase-lite-java-forestdb] No build 'spec' defined, use 'android' by default
[couchbase-lite-java-sqlcipher] No build 'spec' defined, use 'android' by default
build 'couchbase-lite-android-liteserv'
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':couchbase-lite-android-liteserv'.
> Module version couchbase-lite-android-liteserv.libraries:couchbase-lite-java-javascript:unspecified depends on libraries but is not a library itself

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

```

**Comment by andreibaranouski** [comment]
Hi @hideki 
do you have estimation when we could use liteServ project again?

**Comment by pasin** [comment]
This should be done within this week. I expect to work on this tomorrow.

**Comment by pasin** [comment]
Meanwhile, can you use 1.1.0 release branch to unblock this issue?

**Comment by andreibaranouski** [comment]
Thanks @pasin for the update
I'm not planning to change jenkins slave/jobs to use 1.1.0 because it take some time to setup env but I'm able to run our functional tests with 1.1.0 locally on my host.

**Comment by pasin** [comment]
@andreibaranouski couchbase-lite-java-core build is now the same as before. Could you please try and see if the issue is gone? Thanks!

**Comment by pasin** [comment]
I just ran the test locally with extra/jenkins_build/run_android_liteserv.sh and I was able to install the app on an emulator and ran the lite serv app.

I don't know if `submodule update --init --recursive` is needed or not for building the project but I somehow did that as I normally do with the other repos. Anyway updating recursively takes fairly long time. 

I'm closing the issue now. Let open it if you are still seeing the issue.
